### PR TITLE
WP-415: add samples for names and addresses

### PR DESCRIPTION
**Pull request by ehoogerbeets**
Add samples of how to parse and format person names, and how to parse and format mailing addresses. This is for stories WP-415 and WP-416. This change was actually done by Pugalendhi Ganesan and Guruprasad KN in LGSI, but they do not have push permissions, so I am pushing it for them.

### CODE REVIEW

File: samples/NameFormatting.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        {tag: "br"},
+        {kind: "onyx.Groupbox", classes:"onyx-sample-result-box", components: [
+            {kind: "onyx.GroupboxHeader", content: rb.getString("Format result:")},
+            {name: "rtlResult", fit: true, content: "-", style: "padding: 10px"}
+        ]}
+    ],
+ 
+    buttonActivated: function(inSender, inEvent) {
+	
+	if (this.$.length.getActive().content === 'Short') {
```
Comment (aarontam) [inline samples/NameFormatting.js]: This needs to be guarded (or one of the length items needs to be active by default) as this currently results in an exception with no length item being selected.

File: samples/NameFormatting.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
+enyo.kind({
+    name: "ilib.sample.NameFormatting",
+    kind: "FittableRows",
+    classes: "onyx ilib-onyx-sample enyo-fit",
+    
+    components: [
+        {kind: "Scroller", fit: true, components: [
+            {kind: "FittableColumns", components: [
+                /* Header with selecting locale */
+                {kind: "ilib.sample.ChooseLocale", name: "localeSelector", onSelectedLocale: "setLocale"},
```
Comment (aarontam) [inline samples/NameFormatting.js]: The `setLocale` handler is not implemented in the sample, so changing the locale has no effect.

File: samples/NameParsing.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+enyo.kind({
+    name: "ilib.sample.NameParsing",
+    kind: "FittableRows",
+    classes: "onyx ilib-onyx-sample enyo-fit",
+    
+    components: [
+        {kind: "Scroller", fit: false, components: [
+            {kind: "FittableColumns", components: [
+                /* Header with selecting locale */
+                {kind: "ilib.sample.ChooseLocale", name: "localeSelector", onSelectedLocale: "setLocale"},
```
Comment (aarontam) [inline samples/NameParsing.js]: The `setLocale` handler is not implemented in the sample, so changing the locale has no effect.

Comment (aarontam) [inline samples/NameParsing.js]: I think we can remove `onSelectedLocale: "setLocale"` as this is already defined in `ilib.sample.ChooseLocale`.

File: samples/NameFormatting.js (hunk 2 above)
Comment (aarontam) [inline samples/NameFormatting.js]: I think we can remove `onSelectedLocale: "setLocale"` as this is already defined in `ilib.sample.ChooseLocale`.

**Comment by ehoogerbeets** [comment]
@aarontam , your review comments have been addressed now. Please re-review. Thanks!

**Pull request merged**